src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,25 @@
+import Popup from "./Popup";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector }) {
+    super({ popupSelector });
+    this._popupForm = this._popup.querySelector(".modal__form");
+    this._submitButton = this._popup.querySelector(".modal__button");
+  }
+
+  handleDelete(handleDeleteSubmit) {
+    this._handleDeleteSubmit = handleDeleteSubmit;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleDeleteSubmit();
+    });
+  }
+
+  deleting = (deleting) => {
```

Comment:
Needs correction: The name of the function should begin with a verb, reflecting the action that the function performs. Next, the elements to which the action refers are described. `renderLoading` would be good.